src/udev/udevadm-verify.c

```diff
@@ -119,20 +119,47 @@ static int verify_rules_file(UdevRules *rules, const char *fname) {
         return 0;
 }
 
-static int verify_rules(UdevRules *rules, char **files) {
-        size_t fail_count = 0, success_count = 0;
+static int verify_rules_filelist(UdevRules *rules, char **files, size_t *fail_count, size_t *success_count);
+
+static int verify_rules_dir(UdevRules *rules, const char *dir, size_t *fail_count, size_t *success_count) {
+        int r;
+        _cleanup_strv_free_ char **files = NULL;
+
+        r = conf_files_list(&files, ".rules", NULL, 0, dir);
+        if (r < 0)
+                return log_error_errno(r, "Failed to enumerate rules files: %m");
+
+        return verify_rules_filelist(rules, files, fail_count, success_count);
+}
+
+static int verify_rules_filelist(UdevRules *rules, char **files, size_t *fail_count, size_t *success_count) {
         int r, rv = 0;
 
         STRV_FOREACH(fp, files) {
-                r = verify_rules_file(rules, *fp);
-                if (r < 0) {
-                        fail_count++;
-                        if (rv >= 0)
-                                rv = r;
-                } else
-                        success_count++;
+                struct stat st;
+
+                if (stat(*fp, &st) == 0 && S_ISDIR(st.st_mode))
```

Comment:
We have path_is_dir() in stat-util.sh (the name may be slightly different, as I am now writing this from phone.)